packages/contractkit/src/wrappers/Governance.ts

```diff
@@ -72,4 +148,399 @@ export class GovernanceWrapper extends BaseWrapper<Governance> {
+   */
+  proposalExists: (proposalID: BigNumber.Value) => Promise<boolean> = proxyCall(
+    this.contract.methods.proposalExists,
+    tupleParser(valueToString)
+  )
+
+  /**
+   * Returns the current upvoted governance proposal ID and applied vote weight (zeroes if none).
+   * @param upvoter Address of upvoter
+   */
+  getUpvoteRecord: (upvoter: Address) => Promise<UpvoteRecord> = proxyCall(
+    this.contract.methods.getUpvoteRecord,
+    tupleParser(identity),
+    (o) => ({
+      proposalID: valueToBigNumber(o[0]),
+      upvotes: valueToBigNumber(o[1]),
+    })
+  )
+
+  /**
+   * Returns whether a given proposal is queued.
+   * @param proposalID Governance proposal UUID
+   */
+  isQueued = proxyCall(this.contract.methods.isQueued, tupleParser(valueToString))
+
+  /**
+   * Returns the upvotes applied to a given proposal.
+   * @param proposalID Governance proposal UUID
+   */
+  getUpvotes = proxyCall(
+    this.contract.methods.getUpvotes,
+    tupleParser(valueToString),
+    valueToBigNumber
+  )
+
+  /**
+   * Returns the yes, no, and abstain votes applied to a given proposal.
+   * @param proposalID Governance proposal UUID
+   */
+  getVotes = proxyCall(
+    this.contract.methods.getVoteTotals,
+    tupleParser(valueToString),
+    (res): Votes => ({
+      [VoteValue.Yes]: valueToBigNumber(res[0]),
+      [VoteValue.No]: valueToBigNumber(res[1]),
+      [VoteValue.Abstain]: valueToBigNumber(res[2]),
+    })
+  )
+
+  /**
+   * Returns the proposal queue as list of upvote records.
+   */
+  getQueue = proxyCall(this.contract.methods.getQueue, undefined, (arraysObject) =>
+    zip<string, string, UpvoteRecord>(
+      (_id, _upvotes) => ({
+        proposalID: valueToBigNumber(_id),
+        upvotes: valueToBigNumber(_upvotes),
+      }),
+      arraysObject[0],
+      arraysObject[1]
+    )
+  )
+
+  /**
+   * Returns the proposal dequeue as list of proposal IDs.
+   */
+  getDequeue = proxyCall(this.contract.methods.getDequeue, undefined, (arrayObject) =>
+    arrayObject.map(valueToBigNumber)
+  )
+
+  /**
+   * Dequeues any queued proposals if `dequeueFrequency` seconds have elapsed since the last dequeue
+   */
+  dequeueProposalsIfReady = proxySend(this.kit, this.contract.methods.dequeueProposalsIfReady)
+
+  /**
+   * Returns the number of votes that will be applied to a proposal for a given voter.
+   * @param voter Address of voter
+   */
+  async getVoteWeight(voter: Address) {
+    const lockedGoldContract = await this.kit.contracts.getLockedGold()
+    return lockedGoldContract.getAccountTotalLockedGold(voter)
+  }
+
+  private async getDequeueIndex(proposalID: BigNumber.Value) {
+    const dequeue = await this.getDequeue()
+    const index = dequeue.findIndex((d) => d.isEqualTo(proposalID))
+    if (index === -1) {
+      throw new Error(`Proposal ${proposalID} not in dequeue`)
+    }
+    return index
+  }
+
+  // TODO: merge with SortedOracles/Election findLesserAndGreater
+  // proposalID is zero for revokes
```

Comment:
I see thee appeal to use this method for both votes and revokes, but it IMO made for a quite confusing reading. I would recommend for extracting of much of the logic into helper functions and then have separate functions for votes/revokes.